src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -681,55 +365,32 @@ class FastSync(
       context watch handler
       assignedHandlers += (handler -> peer)
       peerRequestsTime += (peer -> Instant.now())
-      syncState = syncState.copy(
-        pendingNonMptNodes = remainingNonMptNodes,
-        pendingMptNodes = remainingMptNodes)
       requestedMptNodes += handler -> mptNodesToGet
       requestedNonMptNodes += handler -> nonMptNodesToGet
     }
 
     def unassignedPeers: Set[Peer] = peersToDownloadFrom.keySet diff assignedHandlers.values.toSet
 
-    def anythingToDownload: Boolean =
-      syncState.anythingQueued || syncState.bestBlockHeaderNumber < syncState.safeDownloadTarget
-
     def fullySynced: Boolean = {
-      syncState.bestBlockHeaderNumber >= syncState.safeDownloadTarget &&
-      !syncState.anythingQueued &&
-      assignedHandlers.isEmpty
+      !syncState.anythingToDownload && assignedHandlers.isEmpty
     }
   }
 
-  private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
-    val fullBlocks = receivedHashes.flatMap { hash =>
-      for {
-        header <- blockchain.getBlockHeaderByHash(hash)
-        _ <- blockchain.getBlockBodyByHash(hash)
-        _ <- blockchain.getReceiptsByHash(hash)
-      } yield header
-    }
-
-    if (fullBlocks.nonEmpty) {
-      val bestReceivedBlock = fullBlocks.maxBy(_.number)
-      if (appStateStorage.getBestBlockNumber() < bestReceivedBlock.number) {
-        appStateStorage.putBestBlockNumber(bestReceivedBlock.number)
-      }
-    }
-
-  }
 }
 
 object FastSync {
   // scalastyle:off parameter.number
-  def props(fastSyncStateStorage: FastSyncStateStorage, appStateStorage: AppStateStorage, blockchain: Blockchain,
-  validators: Validators, peerEventBus: ActorRef, etcPeerManager: ActorRef, syncConfig: SyncConfig, scheduler: Scheduler): Props =
-    Props(new FastSync(fastSyncStateStorage, appStateStorage, blockchain, validators, peerEventBus, etcPeerManager, syncConfig, scheduler))
+  def props(fastSyncStateStorage: FastSyncStateStorage, appStateStorage: AppStateStorage, fastSyncStateHandler: FastSyncStateHandler,
```

Comment:
formatting should be improved (all params in one column if they're not fitting single row)

Reply:
i think this could be improved in many places :)